pylint/checkers/utils.py

```diff
@@ -1172,9 +1172,14 @@ def class_is_abstract(node: nodes.ClassDef) -> bool:
         if meta.name == "ABCMeta" and meta.root().name in ABC_MODULES:
             return True
 
-    for ancestor in node.ancestors():
-        if ancestor.name == "ABC" and ancestor.root().name in ABC_MODULES:
-            # abc.ABC inheritance
+    # As well as direct abc.ABC inheritance
+    for base in node.bases:
+        inferred_base = safe_infer(base)
+        if (
+            isinstance(inferred_base, nodes.ClassDef)
+            and inferred_base.root().name in ABC_MODULES
+            and inferred_base.name == "ABC"
```

Comment:
```suggestion
            and inferred_base.root().name in ABC_MODULES
            and inferred_base.name == "ABC"
```

I think this works fine. Note that you could also call `.qname()` for a similar result.

Reply:
Yes, but do we not care about checking all the modules in `ABC_MODULES`?